File: Backend_DotNet/WheelShare_dotNet/WheelShare_dotNet/Models/EmailOtp.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WheelShare_dotNet.Models
{
    [Table("EmailOtps")]
    public class EmailOtp
    {

        [Key]
        public int OtpId { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public string OtpCode { get; set; }

        public DateTime ExpiresAt { get; set; }

        public bool IsUsed { get; set; } = false;

        [ForeignKey("UserId")]
        public User User { get; set; }
    }
}

File: Backend_Springboot/cab-booking/src/main/java/com/wheelshare/cabbooking/entity/EmailOtp.java
package com.wheelshare.cabbooking.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "EmailOtps")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EmailOtp {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer otpId;

    @Column(nullable = false)
    private Integer userId;

    @Column(nullable = false)
    private String otpCode;

    private LocalDateTime expiresAt;

    private Boolean isUsed = false;

    @ManyToOne
    @JoinColumn(name = "UserId", insertable = false, updatable = false)
    private User user;
}
